experimental/federated_plugin/federated_plugin/android/src/main/kotlin/dev/flutter/federated_plugin/FederatedPlugin.kt

```diff
@@ -4,37 +4,122 @@
+    private lateinit var result: Result
+    private val REQUEST_CODE = 1001
+
+    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "location")
+        channel.setMethodCallHandler(this)
+        context = flutterPluginBinding.applicationContext
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        this.result = result
+        if (call.method == "getLocation") {
+            // Check for the runtime permission if SDK version is greater than 23. If permission
+            // are granted, send the location data back to Dart. If permission are not granted,
+            // request for the runtime permissions.
+            if (android.os.Build.VERSION.SDK_INT > android.os.Build.VERSION_CODES.LOLLIPOP) {
+                if (checkPermissions()) {
+                    fetchLocation()
+                } else {
+                    requestPermissions(activity, arrayOf(
+                            Manifest.permission.ACCESS_FINE_LOCATION,
+                            Manifest.permission.ACCESS_COARSE_LOCATION
+                    ), REQUEST_CODE)
+                }
+            } else {
+                fetchLocation()
+            }
+        } else {
+            result.notImplemented()
+        }
+    }
+
+    // Method to send the last known location of the device to Dart.
+    private fun fetchLocation() {
+        getFusedLocationProviderClient(context).lastLocation
+                .addOnSuccessListener { location ->
+                    if (location != null) {
+                        result.success(listOf(location.longitude, location.latitude))
+                    } else {
+                        sendError("NOT_DETERMINED", "Not able to determine location")
+                    }
+                }.addOnFailureListener { exception ->
+                    sendError("Error", exception.message!!)
+                }
+    }
+
+    // Method to send an error to the Dart while fetching the last known location.
+    private fun sendError(errorCode: String, message: String) {
+        result.error(errorCode, message, null)
+    }
+
+    // Method to check permissions to access the location data.
+    private fun checkPermissions(): Boolean {
+        val fineLocationPermission = ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION)
+        val coarseLocationPermission = ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION)
+
+        return fineLocationPermission == PackageManager.PERMISSION_GRANTED &&
+                coarseLocationPermission == PackageManager.PERMISSION_GRANTED
+    }
+
+
+    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
+    }
+
+    override fun onDetachedFromActivity() {
+        TODO("Not yet implemented")
```

Comment:
If the code doesn't need to do anything here, just remove the TODO and add a comment to that effect (or remove the override, if that's possible).